samples/snippets/sample_custom_types.py

```diff
@@ -0,0 +1,527 @@
+#  Copyright 2021 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# [START compute_custom_machine_type_create ]
+from collections import namedtuple
+from enum import Enum, unique
+import sys
+import time
+from typing import Union
+
+from google.cloud import compute_v1
+
+
+# [END compute_custom_machine_type_create ]
+
+
+# [START compute_custom_machine_type_helper_class ]
+def gb_to_mb(value: int) -> int:
+    return value << 10
+
+
+class CustomMachineType:
+    """
+    Allows to create custom machine types to be used with the VM instances.
+    """
+
+    @unique
+    class CPUSeries(Enum):
+        N1 = "custom"
+        N2 = "n2-custom"
+        N2D = "n2d-custom"
+        E2 = "e2-custom"
+        E2_MICRO = "e2-custom-micro"
+        E2_SMALL = "e2-custom-small"
+        E2_MEDIUM = "e2-custom-medium"
+
+    TypeLimits = namedtuple(
+        "TypeLimits",
+        [
+            "allowed_cores",
+            "min_mem_per_core",
+            "max_mem_per_core",
+            "allow_extra_memory",
+            "extra_memory_limit",
+        ],
+    )
+
+    LIMITS = {
+        CPUSeries.E2: TypeLimits(range(2, 33, 2), 512, 8192, False, 0),
```

Comment:
This value is an iterator, the others are arrays or, in two cases, sets. Is there a reason for the differences?

Reply:
Turned everything into frozensets.